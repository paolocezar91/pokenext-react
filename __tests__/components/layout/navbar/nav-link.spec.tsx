import NavLink from "@/components/layout/navbar/nav-link";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";

jest.mock("next-intl", () => ({
  useTranslations: () => (key: string) => key,
  useLocale: () => "en",
}));

describe("Nav Link Component", () => {
  it("renders correctly with href, className and children", () => {
    render(
      <NavLink href="/pokedex/bulbasaur" className="bulbasaur">
        Bulbasaur
      </NavLink>,
    );
    const link = screen.getByRole("link");
    expect(link).toBeInTheDocument();
    expect(link).toHaveTextContent("Bulbasaur");
    expect(link).toHaveClass("bulbasaur");
    expect(link).toHaveAttribute("href", "/en/pokedex/bulbasaur");
  });

  it("should match snapshot", () => {
    const { asFragment } = render(
      <NavLink href="/pokedex/bulbasaur">Bulbasaur</NavLink>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
