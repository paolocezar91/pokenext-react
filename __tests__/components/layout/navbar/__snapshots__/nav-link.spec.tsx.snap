// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Link Component should match snapshot 1`] = `
<DocumentFragment>
  <a
    class="
    flex
    px-2
    py-2
    text-sm
    rounded
    transition-colors
    hover:shadow-md
    hover:bg-(--pokedex-red-dark)
    active:bg-white
    active:text-(--pokedex-red-dark) bg-transparent "
    href="/en/pokedex/bulbasaur"
  >
    Bulbasaur
  </a>
</DocumentFragment>
`;
